### PR TITLE
Categorized AJAX calls with log file & Event Source xml import.

### PR DESCRIPTION
**Pull request by maheshwaghmare**
### Change log:
- New: Importing site XML with Event Source (SSE) which keep the import process until XML import complete. IE browser does not support the Event Source (SSE) so we have used alternative Yaffle EventSource.
- New: We have categorize the site import process in several AJAX calls.
- New: Generated the import log file. We'll show this if Only if error occurred.
- Improvement: Validate all the possible errors.

### CODE REVIEW

File: inc/classes/class-astra-sites-importer.php [hunk 1]
```diff
@@ -43,162 +43,176 @@ public static function get_instance() {
...
+				$xml_path = Astra_Sites_Helper::download_file( $wxr_url );
+
+				if ( $xml_path['success'] ) {
+
+					if ( isset( $xml_path['data']['file'] ) ) {
+						$data        = Astra_WXR_Importer::instance()->get_xml_data( $xml_path['data']['file'] );
+						$data['xml'] = $xml_path['data'];
+						wp_send_json_success( $data );
+					} else {
+						wp_send_json_error( __( 'Not able to download the XML file!', 'astra-sites' ) );
```
Comment (Nikschavan) [inline inc/classes/class-astra-sites-importer.php]: `There was an error downloading the XML file.`

File: inc/classes/class-astra-sites.php [hunk 2]
```diff
@@ -213,14 +220,51 @@ public function admin_enqueue( $hook = '' ) {
...
+						'importOptions'           => __( '4) Importing "Options"...', 'astra-sites' ),
+						'importOptionsSuccess'    => __( 'Successfully imported Options!', 'astra-sites' ),
+						'importWidgets'           => __( '5) Importing "Widgets"...', 'astra-sites' ),
+						'importWidgetsSuccess'    => __( 'Successfully imported Widgets!', 'astra-sites' ),
+						'serverConfiguration'     => esc_url( 'https://wpastra.com/docs/?p=1314&utm_source=demo-import-panel&utm_campaign=import-error&utm_medium=wp-dashboard' ),
+						'success'                 => __( 'Site imported successfully! visit : ', 'astra-sites' ),
+						'gettingData'             => __( 'Getting Import Data..', 'astra-sites' ),
+						'importingCustomizer'     => __( 'Importing Customizer Settings..', 'astra-sites' ),
+						'importXMLPreparing'      => __( 'Preparing XML Data...', 'astra-sites' ),
+						'importingXML'            => __( 'Importing XML..', 'astra-sites' ),
```
Comment (Nikschavan) [inline inc/classes/class-astra-sites.php]: `Importing Pages, Posts & Media`

File: inc/classes/class-astra-sites.php [hunk 3]
```diff
@@ -213,14 +220,51 @@ public function admin_enqueue( $hook = '' ) {
...
+						'importXMLSuccess'        => __( 'Successfully imported XML!', 'astra-sites' ),
+						'importOptions'           => __( '4) Importing "Options"...', 'astra-sites' ),
+						'importOptionsSuccess'    => __( 'Successfully imported Options!', 'astra-sites' ),
+						'importWidgets'           => __( '5) Importing "Widgets"...', 'astra-sites' ),
+						'importWidgetsSuccess'    => __( 'Successfully imported Widgets!', 'astra-sites' ),
+						'serverConfiguration'     => esc_url( 'https://wpastra.com/docs/?p=1314&utm_source=demo-import-panel&utm_campaign=import-error&utm_medium=wp-dashboard' ),
+						'success'                 => __( 'Site imported successfully! visit : ', 'astra-sites' ),
+						'gettingData'             => __( 'Getting Import Data..', 'astra-sites' ),
+						'importingCustomizer'     => __( 'Importing Customizer Settings..', 'astra-sites' ),
+						'importXMLPreparing'      => __( 'Preparing XML Data...', 'astra-sites' ),
```
Comment (Nikschavan) [inline inc/classes/class-astra-sites.php]: `Setting up import data`

File: inc/classes/class-astra-sites-importer.php (hunk 1 above)
Comment (maheshwaghmare) [reply inc/classes/class-astra-sites-importer.php]: Updated string.

File: inc/classes/class-astra-sites.php [hunk 4]
```diff
@@ -213,14 +220,51 @@ public function admin_enqueue( $hook = '' ) {
...
 						'DescExpand'           => __( 'Read more', 'astra-sites' ) . '&hellip;',
 						'DescCollapse'         => __( 'Hide', 'astra-sites' ),
 						'responseError'        => __( 'There was a problem receiving a response from server.', 'astra-sites' ),
 						'searchNoFound'        => __( 'No Demos found, Try a different search.', 'astra-sites' ),
 						'importWarning'        => __( "Executing Demo Import will make your site similar as ours. Please bear in mind -\n\n1. It is recommended to run import on a fresh WordPress installation.\n\n2. Importing site does not delete any pages or posts. However, it can overwrite your existing content.\n\n3. Copyrighted media will not be imported. Instead it will be replaced with placeholders.", 'astra-sites' ),
 					),
+					'log'             => array(
+						'installingPlugin'        => __( 'Installing plugin ', 'astra-sites' ),
+						'installed'               => __( 'Successfully plugin installed!', 'astra-sites' ),
+						'installError'            => __( 'Error! While Installing plugin ', 'astra-sites' ),
```
[the remark this answers is not in the document]
Comment (maheshwaghmare) [reply inc/classes/class-astra-sites.php]: Updated string.

File: inc/classes/class-astra-sites.php (hunk 3 above)
Comment (maheshwaghmare) [reply inc/classes/class-astra-sites.php]: Updated string.

File: inc/classes/class-astra-sites.php (hunk 2 above)
Comment (maheshwaghmare) [reply inc/classes/class-astra-sites.php]: Updated string.

**Pull request merged**